draft-ietf-quic-http.md

```diff
@@ -1102,69 +1167,33 @@ allowing time for any in-flight requests (at least one round-trip time), the
-cannot push until it receives a MAX_PUSH_ID frame.  A client that wishes to
-manage the number of promised server pushes can increase the maximum Push ID by
-sending a MAX_PUSH_ID frame as the server fulfills or cancels server pushes.
-
-~~~~~~~~~~  drawing
- 0                   1                   2                   3
- 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-|                          Push ID (i)                        ...
-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-~~~~~~~~~~
-{: #fig-max-push title="MAX_PUSH_ID frame payload"}
-
-The MAX_PUSH_ID frame carries a single variable-length integer that identifies
-the maximum value for a Push ID that the server can use (see
-{{frame-push-promise}}).  A MAX_PUSH_ID frame cannot reduce the maximum Push ID;
-receipt of a MAX_PUSH_ID that contains a smaller value than previously received
-MUST be treated as a connection error of type HTTP_MALFORMED_FRAME.
-
-A server MUST treat a MAX_PUSH_ID frame payload that does not contain a single
-variable-length integer as a connection error of type
-HTTP_MALFORMED_FRAME.
-
+An HTTP/QUIC implementation can explicitly close the QUIC connection at any
+time. This results in sending a QUIC APPLICATION_CLOSE frame to the peer; the
+error code in this frame indicates to the peer why the connection is being
+closed.  See {{errors}} for error codes which can be used when closing a
+connection.
 
-# Connection Management
+Before closing the connection, a GOAWAY MAY be sent to allow the client to retry
+requests.  Including the GOAWAY frame in the same packet as the QUIC
+APPLICATION_CLOSE frame improves the chances of the frame being received by
+clients.
 
-QUIC connections are persistent.  All of the considerations in Section 9.1 of
-{{?RFC7540}} apply to the management of QUIC connections.
+## Transport Closure
 
-HTTP clients are expected to use QUIC PING frames to keep connections open.
-Servers SHOULD NOT use PING frames to keep a connection open.  A client SHOULD
-NOT use PING frames for this purpose unless there are responses outstanding for
-requests or server pushes.  If the client is not expecting a response from the
-server, allowing an idle connection to time out (based on the idle_timeout
-transport parameter) is preferred over expending effort maintaining a connection
-that might not be needed.  A gateway MAY use PING to maintain connections in
-anticipation of need rather than incur the latency cost of connection
-establishment to servers.
+For various reasons, the QUIC transport could indicate to the application layer
+that the connection has terminated.  This might be due to an explicit closure
+by the peer, a transport-level error, or a change in network topology which
+interrupts connectivity.
 
+If a connection terminates without a GOAWAY frame, clients MUST assume
+that any request which was sent might have been processed.
```

Comment:
Maybe mention "in whole or part" when it comes to requests, even a partially sent request can be processed.